internal/auth/configuration.go

```diff
@@ -0,0 +1,437 @@
+package auth
+
+import (
+	"encoding/base64"
+	"net/http"
+	"time"
+
+	"github.com/micro/go-micro/config"
+	"github.com/micro/go-micro/config/source/env"
+	"github.com/mitchellh/mapstructure"
+	"golang.org/x/xerrors"
+)
+
+// DefaultAuthConfig specifies all the defaults used to configure sso-auth
+func DefaultAuthConfig() Configuration {
+	return Configuration{
+		ProviderConfigs: map[string]ProviderConfig{},
+		ClientConfigs: map[string]ClientConfig{
+			"proxy": ClientConfig{},
+		},
+		ServerConfig: ServerConfig{
+			Port:   4180,
+			Scheme: "https",
+			TimeoutConfig: TimeoutConfig{
+				Write:   30 * time.Second,
+				Read:    30 * time.Second,
+				Request: 45 * time.Second,
+			},
+		},
+		SessionConfig: SessionConfig{
+			SessionLifetimeTTL: (30 * 24) * time.Hour,
+			CookieConfig: CookieConfig{
+				Expire:   (7 * 24) * time.Hour,
+				Name:     "_sso_auth",
+				Secure:   true,
+				HTTPOnly: true,
+			},
+		},
+		LoggingConfig: LoggingConfig{
+			Enable: true,
+			Level:  "info",
+		},
+		MetricsConfig: MetricsConfig{
+			StatsdConfig: StatsdConfig{
+				Port: 8125,
+				Host: "localhost",
+			},
+		},
+		// we provide no defaults for these right now
+		AuthorizeConfig: AuthorizeConfig{
```

Comment:
I'm not entirely sure what's best for this name - so I don't have strong opinions, but wanted to throw it out there: would `AuthenticatorConfig` be clearer? I'm largely basing that off where it's passed into though.

Reply:
Hmm, after looking through again that doesn't seem right!